scripts/release.sh

```diff
@@ -316,6 +338,10 @@ while test $# -gt 0; do
             help
             exit 0
             ;;
+          --current-branch)
```

Comment:
Should we have a validation that prevents using incompatible flags/environment variables?

For example 
* `--current-branch` should only be called with `DRY_RUN=true`.
* `BRANCH` should be forbidden when `--current-branch` is set
* `REPO` should be forbidden when `--current-branch` is set

Reply:
> --current-branch should only be called with DRY_RUN=true

They are two different concepts. `DRY_RUN` means no push commit & image and no upload. --current-branch means no clone and no checkout. 

> BRANCH should be forbidden when --current-branch is set

Already explained above.

> REPO should be forbidden when --current-branch is set

The same as BRANCH, we need to output the repo in log.

Reply:
> You should not be allowed to upload artifacts from local changes.

This is a valid point. thx & updated.